File: src/main/java/de/tomjuri/armageddon/util/AngleUtil.java
package de.tomjuri.armageddon.util;

import net.minecraft.util.BlockPos;
import net.minecraft.util.MathHelper;

public class AngleUtil {
    public static RotationUtil.Rotation getRotationForBlock(BlockPos targetPos) {
        double diffX = to.xCoord - from.xCoord;
        double diffY = to.yCoord - from.yCoord;
        double diffZ = to.zCoord - from.zCoord;
        double dist = Math.sqrt(diffX * diffX + diffZ * diffZ);

        float pitch = (float) -Math.atan2(dist, diffY);
        float yaw = (float) Math.atan2(diffZ, diffX);
        pitch = (float) wrapAngleTo180((pitch * 180F / Math.PI + 90) * -1);
        yaw = (float) wrapAngleTo180((yaw * 180 / Math.PI) - 90);

        return new Rotation(pitch, yaw);
    }
}

File: src/main/kotlin/de/tomjuri/gemdigger/util/AngleUtil.kt
package de.tomjuri.gemdigger.util

import net.minecraft.util.BlockPos
import kotlin.math.atan2

object AngleUtil {
    fun getLookAtAngles(targetPos: BlockPos, playerX: Double, playerY: Double, playerZ: Double): Pair<Float, Float> {
        val deltaX = targetPos.x + 0.5 - playerX
        val deltaY = targetPos.y + 0.5 - playerY
        val deltaZ = targetPos.z + 0.5 - playerZ
        val distanceXZ = Math.sqrt(deltaX * deltaX + deltaZ * deltaZ)
        val yaw = Math.toDegrees(atan2(deltaZ, deltaX)).toFloat() - 90.0f
        val pitch = -Math.toDegrees(atan2(deltaY, distanceXZ)).toFloat()
        return Pair(yaw, pitch)
    }
}